internal/plugin/dataplane_server.go

```diff
@@ -0,0 +1,153 @@
+/**
+ * Copyright (c) F5, Inc.
+ *
+ * This source code is licensed under the Apache License, Version 2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package plugin
+
+import (
+	"context"
+	"log/slog"
+	"net"
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/nginx/agent/v3/internal/api/http/common"
+	"github.com/nginx/agent/v3/internal/api/http/dataplane"
+	"github.com/nginx/agent/v3/internal/bus"
+	"github.com/nginx/agent/v3/internal/model/instances"
+	"github.com/nginx/agent/v3/internal/model/os"
+	"github.com/nginx/agent/v3/internal/service"
+
+	sloggin "github.com/samber/slog-gin"
+)
+
+const (
+	internalServerError = "Internal Server Error"
+)
+
+type (
+	ErrorResponse struct {
+		Message string `json:"message,omitempty"`
+	}
+
+	DataplaneServer struct {
+		address         string
+		logger          *slog.Logger
+		instanceService service.InstanceServiceInterface
+		messagePipe     *bus.MessagePipe
+		server          net.Listener
+		processes       []*os.Process
+	}
+)
+
+type DataplaneServerParameters struct {
+	Address         string
+	Logger          *slog.Logger
+	instanceService service.InstanceServiceInterface
+}
+
+func NewDataplaneServer(dataplaneServerParameters *DataplaneServerParameters) *DataplaneServer {
```

Comment:
Why is `DataplaneServer` a plugin?

Reply:
gives us the flexibility of disabling the API in future depending on what plugins you specify to enable in your agent config